nat-lab/tests/utils/testing.py

```diff
@@ -10,8 +12,102 @@ def unpack_optional(opt: Optional[T]) -> T:
+    if test_name is not None:
+        try:
+            # take the file path part of the pytest string, remove extension and parent directory
+            file_part = test_name.split("::")[0].split(".")[0].split("/")[-1]
+            return format_path_string(file_part)
+        except IndexError:
+            print(datetime.now(), "get_current_test_file_name() IndexError", test_name)
+            return test_name
+    print(datetime.now(), "PYTEST_CURRENT_TEST is None")
+    return None
+
+
+def get_current_test_case_name() -> str | None:
+    """
+    Returns a string representing the case name of the currently running test.
+
+    The result is derived from PYTEST_CURRENT_TEST env variable.
+
+    For example:
+    "tests/test_mesh_plus_vpn.py::test_vpn_plus_mesh_over_direct" -> "test_vpn_plus_mesh_over_direct"
+    """
     test_name = os.environ.get("PYTEST_CURRENT_TEST")
     if test_name is not None:
-        return "".join([x if x.isalnum() else "_" for x in test_name.split(" ")[0]])
-    return test_name
+        try:
+            test_part = test_name.split("::", 1)[-1].split(" ")[0].split("[", 1)[0]
+            return format_path_string(test_part)
+        except IndexError:
+            print(datetime.now(), "get_current_test_case_name() IndexError", test_name)
+            return test_name
+    print(datetime.now(), "PYTEST_CURRENT_TEST is None")
+    return None
+
+
+def get_current_test_parameter_name() -> str | None:
+    """
+    Returns an optional string representing the parameters of the currently running test.
+
+    The result is derived from PYTEST_CURRENT_TEST env variable.
+
+    For example:
+    "[CONE_CLIENT_2-Default-local-stun-CONE_CLIENT_1-LinuxNativeWg-local-stun]" -> "CONE_CLIENT_2_Default_local_stun_CONE_CLIENT_1_LinuxNativeWg_local_stun"
+    """
+    test_name = os.environ.get("PYTEST_CURRENT_TEST")
+    if test_name is not None:
+        try:
+            parameter_part = test_name.split("::")[-1].split(" ")[0].split("[", 1)[1]
+            return format_path_string(parameter_part)
+        except IndexError:
+            print(
+                datetime.now(),
+                "get_current_test_parameter_name() IndexError",
+                test_name,
+            )
+            return None
+    print(datetime.now(), "PYTEST_CURRENT_TEST is None")
+    return None
+
+
+def format_path_string(input_str: str) -> str:
+    """
+    Format the input string to be safely used as a system path or file name.
+    """
+    # truncate if it's longer than 64 characters to limit long absolute paths
+    # replace any non-alphanumeric characters, and the last "_" if it is present
+    return re.sub(r"\W+", "_", input_str[:64]).rstrip("_")
+
+
+def get_current_test_log_path(base_dir: str) -> str:
+    """
+    Returns a path to the log files of the currently running test.
+    """
+    # format the base_dir
+    if base_dir == "":
+        base_dir = "logs"
+    else:
+        base_dir = format_path_string(base_dir)
```

Comment:
Good idea, I totally agree, but then it can not be re-used for `coredumps` for example in case we decide to use the same structure.. but then we can just put it together in the logs folder 🤔

Reply:
I'm usually a fan of a strategy: add it when it's actually needed